### ISSUE TITLE
HallC hcana Error: Error <THaVform::Compile>: Unknown name :[0]  

### ISSUE THREAD

**Issue by Yero1990** [JeffersonLab/analyzer#132]
Running the Hall C analyzer, the error "Error <THaVform::Compile>: Unknown name :[0] " was encountered. This error was produced for every histogram defined using the "eye" variable, as I commented them out, and the error went away. I compared the histogram definition with an example from Bob Michaels, to make sure it was not a syntax error.

A link to the def file I used when getting the error is below:
https://github.com/JeffersonLab/hallc_replay/blob/master/DEF-files/SHMS/HGCER/phgcerana.def @rwmichaels @

**Comment by rwmichaels** [comment]
It all worked for me at one point.  Reasons not to work are usually a wrong syntax
or some surrounding code changed (e.g. if THaFormula was rewritten).  I'd need to see
the details to comment further.

Bob

----- Original Message -----

**Comment by hansenjo** [comment]
It's interesting that the code seems to think there's a "[0]" somewhere.
There doesn't seem to be a definition like that anywhere in the def file.

There have been no changes to THaFormula since March 2016. The last
changes in THaOutput were Bob's in April 2017.

Which version of ROOT are you using the analyzer with?

Ole

Am 07.07.17 um 12:59 schrieb Robert Michaels:

**Comment by hallc-online** [comment]
I am getting these errors both with ROOT 5.34.36 on cdaql1 and 6.10.00 on my local machine.  I have been playing around with the DEF-files in an attempt to locate the problem with no luck so far.

-Eric

**Comment by pooser** [comment]
Whoops!  I forgot who I was on Git :-p

-Eric

**Comment by rwmichaels** [comment]
It's a dumb question, but do you get a valid looking histogram ?  I mean there's an error message but maybe it does what it should anyway ?  

I'll try to reproduce this sometime but not today.

Bob

----- Original Message -----

**Comment by pooser** [comment]
The histograms appear to be valid.  I would argue that there is not really an "error" per se, although the error is printed to screen at run time.  At the least we need to understand what is causing the error and have it not print to screen regarding this particular corner case.

**Comment by Yero1990** [comment]
Hi Bob,

Yes the histograms are displayed, even with that error appearing.

Carlos

**Comment by rwmichaels** [comment]
I agree we need to understand the error and eliminate it, but now I think I did see it before (gulp!) but got fooled by the good looking histograms, and I think I did some detailed checks that it was filling ok, too.  Even so, I should have chased the suspicous messages at that time.  (Sort of like how you should never publish a code that has compiler warnings).  Sorry about that.  I'll try to fix it.

Bob

----- Original Message -----

**Comment by rwmichaels** [comment]
Ok, I think I fixed this with my commit/pull-request today.  More details
below. Probably too many details considering what a trivial non-problem
it was !  Also, I claim there were no consequences to the error message
and that after the code is fixed the error message disappears and the
output is identical.  I tested it on cdaql1 on a data file there, but
I hope Carlos will check it, too.

    https://github.com/JeffersonLab/analyzer/pull/134

Bob
